settings.gradle.kts

```diff
@@ -173,3 +173,25 @@ FeaturePreviews.Feature.values().forEach { feature ->
         enableFeaturePreview(feature.name)
     }
 }
+
+fun remoteBuildCacheEnabled(settings: Settings) = settings.buildCache.remote?.isEnabled == true
+
+fun isAdoptOpenJDK() = true == System.getProperty("java.vendor")?.let { it.contains("AdoptOpenJDK") || it.contains("Adoptium") }
+
+fun isAdoptOpenJDK11() = isAdoptOpenJDK() && JavaVersion.current().isJava11
+
+fun getBuildJavaHome() = System.getProperty("java.home")
+
+gradle.settingsEvaluated {
+    if ("true" == System.getProperty("org.gradle.ignoreBuildJavaVersionCheck")) {
+        return@settingsEvaluated
+    }
+
+    if (!isAdoptOpenJDK11()) {
```

Comment:
We still use adopt open JDK for building Gradle on CI, right? Should we change this to be Temurian?

I suppose this change would also break folks on Apple M1, since they'd probably use a different vendor.

Reply:
These two questions are actually the same: which vendor shall we consider as supported for remote cache enabled? I guess AdoptOpenJDK and temurin are considered as same because they're in the same series, like AdoptOpenJDK 11.0.9 -> Adoptium 11.0.10? I have no idea how we manage M1 JDK...

Reply:
Ah, Adoptium 11 doesn't support M1 yet...

Reply:
Yeah, I'd allow both vendors. Or shall we only check for the Java version now? So you are sure to run the build with Java 11 and everything else doesn't matter much?

Reply:
We already used termurian as build JDK: https://github.com/gradle/gradle/commit/dd7704328fd4aea8cda2d524ac15dc6aa5b6a26c

So the only problem is that it might break on M1 mac. I don't know how many people are using M1 so I'd just merge it and see how much complaint we get...